### PR TITLE
Review fixes by MM

### PR DESCRIPTION
**Pull request by manojmalik20**
Memoized the getCustomStatus and getRecentCustomStatuses selector
Changed the name of parameters in position utils and modified handling topOffset in emoji picker tabs
Removed the dialogProps from the custom status modal oepening function everywhere
Removed the use of strong tags and used span with CSS
Added localization in the custom status modal and respective ids in i18n/en.json
Added placeholder in quick input component
Updated snapshot of Profile popover and tests of custom status selectors

#### Related Pull Requests

- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

### CODE REVIEW

File: components/custom_status/custom_status_modal.tsx [hunk 1]
```diff
@@ -45,12 +45,17 @@ const defaultCustomStatusSuggestions: DefaultUserCustomStatus[] = [
...
     const dispatch = useDispatch();
-    const currentCustomStatus = useSelector((state: GlobalState) => getCustomStatus(state));
+    let currentCustomStatus = useSelector((state: GlobalState) => getCustomStatus(state));
     const recentCustomStatuses = useSelector((state: GlobalState) => getRecentCustomStatuses(state));
     const customStatusControlRef = useRef(null);
+    const {formatMessage} = useIntl();
     const [showEmojiPicker, setShowEmojiPicker] = useState<boolean>(false);
+    if (!currentCustomStatus) {
+        currentCustomStatus = {};
+    }
```
Comment (chetanyakan) [inline components/custom_status/custom_status_modal.tsx]: Should we merge this with the declaration on line 51?

File: components/post_view/post_header/post_header_custom_status.tsx [hunk 2]
```diff
@@ -44,24 +45,24 @@ const PostHeaderCustomStatus = (props: ComponentProps) => {
...
-                <EmojiIcon className='post__header-set-custom-status-icon'/>
-                <span className='post__header-set-custom-status-text'>
-                    <FormattedMessage
-                        id='post_header.update_status'
-                        defaultMessage='Update your status'
-                    />
-                </span>
-            </div>
-        );
+    if (!(customStatusEnabled && !isCustomStatusSet && showUpdateStatusButton && !isSystemMessage && isCurrentUserPost)) {
```
Comment (chetanyakan) [inline components/post_view/post_header/post_header_custom_status.tsx]: 1. Please return null using a check: 
```js
if (!customStatusEnabled || isSystemMessage) {
    return null;
}
```
Do this before the if condition at line 31/32

2. Should we check `isCurrentUserPost` only using CSS?

File: components/profile_popover/profile_popover.scss [hunk 3]
```diff
@@ -4,3 +4,7 @@
     background: white;
     padding: 0
 }
+
+.bold {
```
Comment (chetanyakan) [inline components/profile_popover/profile_popover.scss]: Don't create utility CSS class names.
```suggestion
.user-profile-popover__heading {
```

Also, update the usages.

File: components/profile_popover/profile_popover.jsx [hunk 4]
```diff
@@ -291,11 +290,11 @@ class ProfilePopover extends React.PureComponent {
                 <div>
                     <button
                         className='user-popover__set-custom-status-btn'
-                        aria-label={Utils.localizeMessage('user_profile.custom-status.set-status', 'Set a status').toLowerCase()}
+                        aria-label={Utils.localizeMessage('user_profile.custom_status.set_status', 'Set a status').toLowerCase()}
```
Comment (chetanyakan) [inline components/profile_popover/profile_popover.jsx]: This `aria-label` repeats the message already visible on the screen.
I think this is not needed as screen readers will read the other message and we can remove this.

Refer: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute

***

On a separate note, should we use `aria-labelled-by` or some other similar tags?

File: selectors/views/custom_status.ts [hunk 5]
```diff
@@ -3,34 +3,37 @@
...
     return config && config.EnableCustomUserStatuses === 'true';
 }
 
 function showCustomStatusPulsatingDotAndPostHeader(state: GlobalState) {
-    const preferences = getMyPreferences(state);
-    const key = getPreferenceKey(Preferences.CATEGORY_CUSTOM_STATUS, Preferences.NAME_CUSTOM_STATUS_TUTORIAL_STATE);
-    const hasViewedCustomStatusModal = preferences[key] && preferences[key].value === Preferences.CUSTOM_STATUS_MODAL_VIEWED;
-    return !hasViewedCustomStatusModal;
+    const hasViewedCustomStatusModal = get(state, Preferences.CATEGORY_CUSTOM_STATUS, Preferences.NAME_CUSTOM_STATUS_TUTORIAL_STATE);
+    return hasViewedCustomStatusModal !== Preferences.CUSTOM_STATUS_MODAL_VIEWED;
```
Comment (chetanyakan) [inline selectors/views/custom_status.ts]: This looks really suspicious. maybe we need to tweak some variable names here.

**Review by chetanyakan — CHANGES_REQUESTED** [review]

File: selectors/views/custom_status.ts (hunk 5 above)
Comment (manojmalik20) [reply selectors/views/custom_status.ts]: This was suggested in the PR comments.

Comment (chetanyakan) [reply selectors/views/custom_status.ts]: What does the `hasViewedCustomStatusModal` variable name mean according to you and how is it used here?

Comment (manojmalik20) [reply selectors/views/custom_status.ts]: I have renamed this variable to customStatusTutorialState

**Review by chetanyakan — APPROVED** [review]

**Pull request merged**